chiselFrontend/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -95,6 +106,27 @@ private[chisel3] trait HasId {
   private[chisel3] def setRef(parent: HasId, index: Int): Unit = setRef(Index(Node(parent), ILit(index)))
   private[chisel3] def setRef(parent: HasId, index: UInt): Unit = setRef(Index(Node(parent), index.ref))
   private[chisel3] def getRef: Arg = _ref.get
+
+  // Implementation of public methods.
+  def signalName = _parent match {
+    case None => getRef.name
```

Comment:
myTopLevelModule.signalName is the only case this should happen, but presumably that should be a legal thing to ask.